File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/di/usecases/UseCasesModule.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.di.usecases;


import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.Impl.TryToLoginUseCaseImpl;
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.TryToLoginUseCase;

import dagger.Module;
import dagger.Provides;

@Module
public class UseCasesModule {

    @Provides
    public TryToLoginUseCase provideTryToLoginUseCase(TryToLoginUseCaseImpl tryToLoginUseCaseImpl) {
        return tryToLoginUseCaseImpl;
    }

}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/di/usecases/UseCasesModule.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.di.usecases

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.TryToLoginUseCase
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.impl.TryToLoginUseCaseImpl
import dagger.Module
import dagger.Provides

@Module
class UseCasesModule {

    @Provides
    fun provideTryToLoginUseCase(tryToLoginUseCaseImpl: TryToLoginUseCaseImpl): TryToLoginUseCase {
        return tryToLoginUseCaseImpl
    }
}
